File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        #0.-Store number of instances
        result = []
        #1.- Iterate through List
        for i in range(len(nums)):
            count = 0
            for j in range(len(nums)):
                #2.-Compare current number with others
                if nums[j] < nums[i]:
                    #increase count when its smaller
                    count += 1
            #4.-add results to new array
            result.append(count)
        return result
        
        

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        //1.-clone array and sort it 
        int [] sortedNumbers = nums.clone();
        Arrays.sort(sortedNumbers);

        //2.-HaspMap to store(Number, SmallerNumber)
        Map<Integer, Integer> map = new HashMap<>();

        //3.-Fill Map
        for (int i = 0; i < sortedNumbers.length; i++){
            if (!map.containsKey(sortedNumbers[i])){
                map.put(sortedNumbers[i], i);
            }
        }
        //4.-Build new result
        int[] result = new int[nums.length];
        for (int i = 0; i < nums.length; i++){
            result[i] = map.get(nums[i]);
        }
        return result;
    }

}